### DISCUSSION TITLE
How can i install apps from cashe ?

### DISCUSSION THREAD

**Discussion by issam-seghir** [ScoopInstaller/Scoop#5542] [Help and Questions] [answered]
I reset my PC last week, which resulted in the uninstallation of all my apps. 
However, prior to that, I had backed up my .scoop folder on an external drive. 

Currently, my internet connection is quite slow, and I would like to install the old apps from the cached folder instead of downloading them again over my network connection. 

How can I accomplish this? I attempted to copy the cached folder from my backup and paste it into the scoop/cache folder after installing Scoop. However, when I tried to download an app using scoop install app, it still downloaded the app through my network instead of using the cached files 

and if you have a more efficient method to back up my Scoop apps, please let me know

**Comment by HUMORCE** [comment]
> However, prior to that, I had backed up my .scoop folder on an external drive.

Entire `~\scoop`? then just overwrite to `~\scoop` and execute `scoop reset *`

**Reply by issam-seghir** [reply]
What if I change the installation location of Scoop after the Windows reset? Currently, my Scoop folder is located at: 'D:\Apps\scoop'

**Reply by HUMORCE** [reply, marked as answer]
1. Add `D:\Apps\scoop\shims` to PATH
2. Select one of them:
   - Add environment variable 'SCOOP' and set its value to `D:\Apps\scoop`
   - Execute `scoop config ROOT_PATH 'D:\Apps\scoop'`
3. Execute `scoop reset *` to recover environment variables, shims, shortcuts, etc. of apps.

https://github.com/ScoopInstaller/Scoop/blob/f93028001fbe5c78cc41f59e3814d2ac8e595724/lib/core.ps1#L1258-L1260


---
**Optional**

To avoid permission issues, I recommend that you copy the scoop dir and then delete the original one.

```
cp D:\Apps\scoop D:\Apps\scoop-new -recurse
mv D:\Apps\scoop D:\Apps\scoop-backup
mv D:\Apps\scoop-new D:\Apps\scoop

# delete `D:\Apps\scoop-backup` when everything ok.
```
